### DISCUSSION TITLE
Train model with BACMMAN

### DISCUSSION THREAD

**Discussion by sarthursss** [jeanollion/bacmman#15] [General]
Hello,

We recently tried to use your algorithm DiSTNET-2D for track and seg cancer cells for research, because we know that this algorithms is very powerful and useful . So we went to Run > Train Neural Network, and then we got lost instantly. 

Maybe you can point us to a tutorial to train our own model, or explain us the different buttons presents on this page.

Sincerely,

Arthur Salvador 
Intern at IJCLAB, Orsay

**Comment by jannesantana** [comment]
Hey, 
I'm a PhD student at CYU. I'm also very interested in how to use the model training tool. It would be great if we could have a tutorial!
In addition, I'm also interested in understanding better how to fine tune the parameters and which one is responsible for each effect/problems in the tracking.

**Comment by jeanollion** [comment]
Hello Arthur and Janne, 
We are aware that documentation on this point is missing. We will try to write it in July. 
In the meanwhile, please find a few guidelines below. 

1. To train, it is required to have a GPU, and docker installed with GPU pass-though:  https://github.com/jeanollion/bacmman/wiki/Installation#docker
2. Generate a curated training set.  here are a few resources:
    - data visualization:     https://github.com/jeanollion/bacmman/wiki/Visualization
    - data curation: https://github.com/jeanollion/bacmman/wiki/Data-Curation
    - create a selection of the parent object class (that can be viewfield in the case of the examples from DiSTNet, but could also be  microchannels in the case of microfluidic devices) that contains only curated objects. For tracking applications such as DiSTNet2D, try to have long contiguous parent tracks. See : https://github.com/jeanollion/bacmman/wiki/Selections    
3. When data is curated, open the training tab from the run menu, choose a working directory on the right panel and choose DiSTNet2DTraining as the method on the left. On the right panel you have a panel to export a training set as hdf5 file. 
4. Configure training parameters (this will require more documentation) choose the exported dataset file in the dataset list, setup data augmentation, model architecture parameters (in particular number of downsampling, so that objects in the downsampled image are as small as possible but still occupy a few pixels. Downsamling is is 2 for the bacterial example and 3 for the eukaryotic cells example) etc. Data augmentation can be tested with the dedicated button. 
5. start training. when training is finished model needs to be exported. Currently the tensorflow 2.7.1 image needs to be used for export to be able to use the saved model for predictions in bacmman. 

Please also note that I work as an independent ( https://www.sabilab.fr/ ), I'm happy to provide help on documentation and bugfixes but i cannot provide dedicated troubleshooting. I can also provide service for specific developments, training sessions etc... 
Best regards, 
Jean

**Reply by sarthursss** [reply]
Thanks !

Have a good day,

Arthur
